algorithms/C/searching/Binary-search.c

```diff
@@ -0,0 +1,74 @@
+/*
+Problem: Search data using Binary Search
+Binary Search: It is a way to search data in efficient time.It
+works by repeatedly dividing in half the portion of the
+list that could contain the item, until you have narrowed down
+the possible locations to just one.
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+
+//Function to search data using binary search
+int binary_search(int *arr,int size,int to_search)
+{
+    //initialize start as 0 and end as size-1
+    int start=0,end=size-1,mid;
+    //until start is less than or equal to end run the loop
+    while(start<=end)
+    {
+       //every time assign (start+end)/2(middle element of the current array) in mid
+       mid=(start+end)/2;
```

Comment:
While calculating mid-value use `mid = start + ((end - start) / 2);`
As the approach used by you would fail the test cases on the large number but the above approach won't as it won't let the number get out of scope.